# Bonus: Temperature Analysis I

In [1]:
import pandas as pd
from datetime import datetime as dt

In [2]:
# "tobs" is "temperature observations"
df = pd.read_csv('Resources/hawaii_measurements.csv')
df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [3]:
# Convert the date column format from string to datetime
df.date = pd.to_datetime(df.date, infer_datetime_format=True)

In [4]:
# Set the date column as the DataFrame index
df = df.set_index(df['date'])
df.head()

station       date  prcp  tobs
date                                          
2010-01-01  USC00519397 2010-01-01  0.08    65
2010-01-02  USC00519397 2010-01-02  0.00    63
2010-01-03  USC00519397 2010-01-03  0.00    74
2010-01-04  USC00519397 2010-01-04  0.00    76
2010-01-06  USC00519397 2010-01-06   NaN    73

In [5]:
# Drop the date column
df = df.drop(columns='date')
df.head()

station  prcp  tobs
date                               
2010-01-01  USC00519397  0.08    65
2010-01-02  USC00519397  0.00    63
2010-01-03  USC00519397  0.00    74
2010-01-04  USC00519397  0.00    76
2010-01-06  USC00519397   NaN    73

### Compare June and December data across all years 

In [6]:
from scipy import stats

In [7]:
# Filter data for desired months
jun_data = df[df.index.month == 6]
dec_data = df[df.index.month == 12]

In [8]:
# Identify the average temperature for June
jun_data.mean()

prcp     0.136360
tobs    74.944118
dtype: float64

In [9]:
# Identify the average temperature for December
dec_data.mean()

prcp     0.216819
tobs    71.041529
dtype: float64

In [10]:
# Create collections of temperature data
june_temp = jun_data.tobs
dec_temp = dec_data.tobs

In [11]:
# Run paired t-test
stats.ttest_ind(june_temp, dec_temp)

Ttest_indResult(statistic=31.60372399000329, pvalue=3.9025129038616655e-191)

### Analysis
All the stations, the mean temperatures in June and December in the years 2010-2017 differby 3.9 degrees celsius. The unpaired t-test results in an low p-vale, the difference is statistically signicficant.